homeassistant/components/plugwise/config_flow.py

```diff
@@ -77,6 +84,29 @@ async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> Smile:
     return api
 
 
+async def verify_connection(
+    hass: HomeAssistant, user_input: dict[str, Any]
+) -> tuple[Smile | None, dict[str, str]]:
+    """Verify and return the gateway connection or an error."""
+    errors: dict[str, str] = {}
+
+    try:
+        return (await validate_input(hass, user_input), errors)
+    except ConnectionFailedError:
+        errors[CONF_BASE] = "cannot_connect"
+    except InvalidAuthentication:
+        errors[CONF_BASE] = "invalid_auth"
+    except InvalidSetupError:
+        errors[CONF_BASE] = "invalid_setup"
+    except (InvalidXMLError, ResponseError):
+        errors[CONF_BASE] = "response_error"
+    except UnsupportedDeviceError:
+        errors[CONF_BASE] = "unsupported"
+    except Exception:  # noqa: BLE001
+        errors[CONF_BASE] = "unknown"
```

Comment:
Ideally we also log the exception when we catch a bare exception

Reply:
[367070e](https://github.com/home-assistant/core/pull/132878/commits/367070ed55acf3a634adde5c0d2ae79e1240b9ed) covers that now (but it won't show up here since we moved that part to a separate function now)